server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourcePhase.java

```diff
@@ -60,8 +60,8 @@ private void hitExecute(FetchSourceContext fetchSourceContext, HitContext hitCon
                 final boolean nestedHit = hitContext.hit().getNestedIdentity() != null;
                 Source source = hitContext.source();
 
-                // If this is a parent document and there are no source filters, then add the source as-is.
-                if (nestedHit == false && fetchSourceContext.hasFilter() == false) {
+                // If this is a parent document and there are no source filters and include_vector is set as true, then add the source as-is.
+                if (nestedHit == false && fetchSourceContext.hasFilter() == false && fetchSourceContext.filterVectorFields() == Boolean.FALSE) {
```

Comment:
Better to handle this case like so:

```suggestion
                if (nestedHit == false && sourceFilter.isEmpty()) {
```

And add an `isEmpty` method to `SourceFilter`.

Reply:
Are you suggesting that `include_vectors` should live in the `SourceFilter` class similar to `includes` and `excludes`?

Reply:
No, `SourceFilter` already has a boolean that tracks when it is empty: https://github.com/elastic/elasticsearch/blob/7ed58e75dab2c36c99aabd78157be166c4ec322f/server/src/main/java/org/elasticsearch/search/lookup/SourceFilter.java#L39

Just expose that with a new getter. Any embedding field exclusions are already baked into the `excludes` value passed to `SourceFilter`.